src/js/server/devpmt-morphic-dispatchers.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * Dispatchers for Morphic UI Workflow
+ *
+ * Copyright 2019 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+/* eslint-env node */
+"use strict";
+var fluid = require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+
+// URLPATH /login
+fluid.defaults("gpii.devpmt.morphic.loginToSafeHandler", {
+    gradeNames: ["gpii.devpmt.baseDispatcher"],
+    method: "use",
+    defaultTemplate: "morphic-login",
+    invokers: {
+        middleware: {
+            funcName: "gpii.devpmt.morphic.loginToSafeHandler.handleRequest",
+            args: ["{that}", "{gpii.devpmt}", "{arguments}.0", "{arguments}.1", "{arguments}.2"] // req, res, next
+        }
+    }
+});
+
+gpii.devpmt.morphic.loginToSafeHandler.handleRequest = function (that, devpmt, req, res, next) {
+    if (req.method === "POST") {
+        var unlockProm = devpmt.cloudSafeUnlockDataSource.set({}, {
+            username: req.body.username,
+            password: req.body.password
+        });
+        unlockProm.then(function (data) {
+            req.session.loggedInToSafe = data.id;
+            res.redirect("/morphic/safe");
+        }, function (err) {
+            fluid.log(err);
```

Comment:
Seems the error will only be logged but won't be sent and displayed on UI. Is there  a plan to display the error?

Reply:
Maybe, the plan is not to do any additional UI work or features until after this initial branch of work is tagged as 1.0 and merged in.